src/core/annotation.js

```diff
@@ -1815,86 +1818,115 @@ class WidgetAnnotation extends Annotation {
     let appearance = await this._getAppearance(
       evaluator,
       task,
+      RenderingIntentFlag.SAVE,
       annotationStorage
     );
     if (appearance === null) {
+      // Appearance didn't change.
       return null;
     }
     const { xref } = evaluator;
 
-    const dict = xref.fetchIfRef(this.ref);
-    if (!(dict instanceof Dict)) {
+    const originalDict = xref.fetchIfRef(this.ref);
+    if (!(originalDict instanceof Dict)) {
       return null;
     }
 
-    const bbox = [
-      0,
-      0,
-      this.data.rect[2] - this.data.rect[0],
-      this.data.rect[3] - this.data.rect[1],
-    ];
+    const dict = new Dict(xref);
+    originalDict.forEachRaw((key, val) => {
+      if (key !== "AP") {
+        dict.set(key, val);
+      }
+    });
```

Comment:
Do we really need to add a new method in the `Dict`-class for a single call-site, or could we perhaps just re-factor this code a little bit?
```suggestion
    for (const key of originalDict.getKeys()) {
      if (key !== "AP") {
        dict.set(key, originalDict.getRaw(key));
      }
    }
```